.github/workflows/build-containers.yml

```diff
@@ -1,60 +1,135 @@
+          else
+            containers=$(echo -n ${{ steps.changed-containers.outputs.all_changed_files }} | jq -R -s -c 'split(" ")')
+          fi
+
+          echo Containers to build: $containers
+          echo ::set-output name=containers::$containers
 
-  docker:
-    name: Build and push
-    needs: list-containers
-    runs-on: [self-hosted, 1ES.Pool=azure-osconfig-linux-pool]
+  docker-build:
+    name: Docker build & push
+    needs: modified-containers
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
     strategy:
-        matrix:
-            container: ${{ fromJson(needs.list-containers.outputs.matrix) }}
+      matrix:
+        container: ${{ fromJson(needs.modified-containers.outputs.containers) }}
     steps:
       - uses: actions/checkout@v2
+        with:
+          # Only checkout submodules recursively if this is a pull request
+          submodules: ${{ github.event_name == 'pull_request' && 'recursive' || 'false' }}
 
-      - name: Set image
+      - name: Setup image
         id: image
         run: |
-          DOCKERFILE=${{ matrix.container }}
-          echo "::set-output name=name::$(echo ${DOCKERFILE%/*}-dev)"
-          echo "::set-output name=path::$(dirname ${DOCKERFILE})"
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v1
+          dockerfile=${{ matrix.container }}
+          image=${dockerfile#devops/docker/}
+          image=${image%/*}
+          distro=${image%-*}
+          platform=${image#*-}
+
+          repo=$(echo ${{ github.repository }} | awk '{print tolower($0)}')
+
+          echo ::set-output name=name::$(echo -n $repo/$image)
+          echo ::set-output name=path::$(dirname ${dockerfile})
+          echo ::set-output name=distro::$(echo -n $distro)
+          echo ::set-output name=platform::$(echo -n $platform)
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
 
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
 
-      - name: Login to ACR
-        uses: docker/login-action@v1
+      - name: Docker login
+        uses: docker/login-action@v2
         with:
-          registry: ${{ secrets.ACR_REGISTRY }}
-          username: ${{ secrets.ACR_CLIENT_ID }}
-          password: ${{ secrets.ACR_CLIENT_SECRET }}
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
 
-      - name: Build and push
-        uses: docker/build-push-action@v2
+      - name: Docker build
+        uses: docker/build-push-action@v3
         with:
           context: ${{ steps.image.outputs.path }}
+          load: true
+          tags: |
+            ${{ steps.image.outputs.name }}:latest
+            ${{ steps.image.outputs.name }}:${{ github.run_number }}
+            ${{ steps.image.outputs.name }}:${{ github.sha }}
```

Comment:
is the sha useful as a tag? Can we have a timestamp tag?

Reply:
We could add a timestamp tag, however `sha` and `run_number` are already unique in terms of the container image. Also containers are time stamped by when they are pushed to the registry